File: ios/RnBabbleSdk.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RnBabbleSdk, NSObject)


RCT_EXTERN_METHOD(configure:(NSString *)userId)

RCT_EXTERN_METHOD(setCustomerId:(NSString *)customerId userDetails:(NSDictionary *)userDetails)

RCT_EXTERN_METHOD(triggerSurvey:(NSString *)trigger properties:(NSDictionary *)properties)

RCT_EXTERN_METHOD(cancelSurvey)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/RnBabbleSdk.swift
import BabbleiOS
@objc(RnBabbleSdk)
class RnBabbleSdk: NSObject {
  
    @objc(multiply:withB:withResolver:withRejecter:)
    func multiply(a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
      resolve(a*b)
    }
    
    @objc(configure:)
    func configure(userId: String) -> Void {
        BabbleSdk.configure(userId)
    }
        
    @objc(setCustomerId:userDetails:)
    func setCustomerId(customerId: String,userDetails: [String: Any]? = nil) -> Void {
        BabbleSdk.setCustomerId(customerId, userDetails: userDetails)
    }
    
    @objc(triggerSurvey:properties:)
    func triggerSurvey(trigger: String,properties: [String: Any]? = nil) -> Void {
        DispatchQueue.main.async {
            BabbleSdk.trigger(trigger, properties: properties)
        }
    }
}
